states.c

```diff
@@ -21,75 +22,42 @@
 #include <linux/spinlock.h>
 #include <linux/idr.h>
 #include <linux/ratelimit.h>
+#include <linux/hashtable.h>
+#include <linux/stringhash.h>
 
-#include <lualib.h>
+#include "lua/lualib.h"
 
 #include "luautil.h"
-#include "xt_lua.h"
 #include "states.h"
-#include "netlink.h"
-#include "kpi_compat.h"
-
-#ifndef NFLUA_SETPAUSE
-#define NFLUA_SETPAUSE	100
-#endif /* NFLUA_SETPAUSE */
-
-extern int luaopen_memory(lua_State *);
-extern int luaopen_conn(lua_State *);
-extern int luaopen_netlink(lua_State *);
-extern int luaopen_packet(lua_State *);
-extern int luaopen_timer(lua_State *);
-
-static const luaL_Reg libs[] = {
-	{"memory", luaopen_memory},
-	{"conn", luaopen_conn},
-	{"netlink", luaopen_netlink},
-	{"packet", luaopen_packet},
-	{"timer", luaopen_timer},
-	{NULL, NULL}
-};
 
-static inline int name_hash(void *salt, const char *name)
-{
-	int len = strnlen(name, NFLUA_NAME_MAXSIZE);
-	return kpi_full_name_hash(salt, name, len) & (XT_LUA_HASH_BUCKETS - 1);
-}
-
-static bool refcount_dec_and_lock_bh(kpi_refcount_t *r, spinlock_t *lock)
-{
-	if (kpi_refcount_dec_not_one(r))
-		return false;
+#ifndef KLUA_SETPAUSE
+#define KLUA_SETPAUSE	100
+#endif /* KLUA_SETPAUSE */
 
-	spin_lock_bh(lock);
-	if (!kpi_refcount_dec_and_test(r)) {
-		spin_unlock_bh(lock);
-		return false;
-	}
+static struct meta_state ms;
```

Comment:
This variable still here to implement the unique meta-state to all operations related to states management